src/modules/verifiable-credentials/verifiable-credentials-base.service.ts

```diff
@@ -477,6 +488,28 @@ export abstract class VerifiableCredentialsServiceBase {
     return true;
   }
 
+  /**
+   * Returns issued role verifiable credentials which matches definition
+   *
+   * @param presentationDefinition
+   * @returns results of matching each role verifiable credential to definition
+   */
+  public async getCredentialsByDefinition(
+    presentationDefinition: IPresentationDefinition
+  ): Promise<SelectResults> {
+    const claims = await this._cacheClient.getClaimsBySubject({
+      did: this._signerService.did,
+      isAccepted: true,
+    });
+    const vc = claims
+      .filter((claim) => claim.vp)
+      .map((claim) => claim.vp!.verifiableCredential)
+      .reduce((acc, curr) => [...acc, ...curr], []);
```

Comment:
```suggestion
      .flat();
```
or `flatMap` instead of `map` line earlier

Reply:
Yeah, this is nice. Had to change compilation target to es2019 though. Hope this is safe. @jrhender ?

Reply:
@JGiter , I think the change to es2019 is okay